File: Python/3Sum.py
from typing import List

class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        check = set()
        
        nums.sort()
        target = 0

        for i in range(len(nums)):
            tar = target - nums[i]
            si, ei = i + 1, len(nums) - 1

            while si < ei:
                if nums[si] + nums[ei] > tar:
                    ei -= 1
                elif nums[si] + nums[ei] < tar:
                    si += 1
                else:
                    triplet = (nums[si], nums[ei], nums[i])
                    if triplet not in check:
                        check.add(triplet)
                        ans.append([nums[si], nums[ei], nums[i]])
                    si += 1
                    ei -= 1

        return ans

File: Java/3Sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        List<List<Integer>> ans= new ArrayList<>();
        HashSet<List<Integer>> check= new HashSet<>();

        Arrays.sort(nums);

        int t= 0;
        for(int i=0; i<nums.length; i++){
            int tar= t-nums[i];
            int si= i+1;
            int ei= nums.length-1;

            while(si<ei){
                if(nums[si]+nums[ei]>tar){
                    ei-=1;
                }else if(nums[si]+nums[ei]<tar){
                    si+=1;
                }else{
                    if(!check.contains(Arrays.asList(nums[si], nums[ei], nums[i]))){
                        check.add(Arrays.asList(nums[si], nums[ei], nums[i]));
                        ans.add(Arrays.asList(nums[si], nums[ei], nums[i]));
                    }
                    si+=1;
                    ei-=1;
                }
            }  
        }
        
        return ans;
    }
}